mitmproxy/tools/cmdline.py

```diff
@@ -486,6 +487,11 @@ def proxy_options(parser):
         action="store_true", dest="spoof_source_address",
         help="Use the client's IP for server-side connections"
     )
+    group.add_argument(
+        "--upstream-bind-address",
+        action="store", type=str, dest="upstream_bind_address", default='',
+        help="Address to bind upstream requests to (defaults to none)"
+    )
```

Comment:
Can/should we make this mutually excluse with spoof_source_address?
(see `websockets` just above this for how)